iceoryx_utils/source/posix_wrapper/semaphore.cpp

```diff
@@ -79,27 +81,33 @@ void Semaphore::closeHandle() noexcept
     }
 }
 
-bool Semaphore::getValue(int& value) const noexcept
+cxx::expected<int, SemaphoreError> Semaphore::getValue() const noexcept
 {
-    return m_isInitialized
-           && !cxx::makeSmartC(iox_sem_getvalue, cxx::ReturnMode::PRE_DEFINED_ERROR_CODE, {-1}, {}, m_handlePtr, &value)
-                   .hasErrors();
-}
+    int value;
```

Comment:
Shouldn't we prefer fixed width integer types?

Reply:
In general you are correct but the posix semaphore explicitly contains an integer therefore we have to stick to that type for now.

Reply:
Couldn't we then hit issues with platform independence? TBH I'd prefer a cast.